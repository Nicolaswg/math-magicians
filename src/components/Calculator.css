*,
*::before,
*::after {
  box-sizing: border-box;
}

.calculator-grid {
  display: flex;
  flex-wrap: wrap;
  flex: 0 1 60%;
  min-width: 350px;
  max-width: 450px;
}

.output {
  height: 60px;
  display: flex;
  justify-content: flex-end;
  align-items: center;
  padding: 10px;
  background-color: rgb(126, 126, 126);
  flex: 0 1 100%;
  border: 1px solid black;
  color: #fff;
  font-size: 1.5rem;
}

.row {
  display: flex;
  flex-wrap: wrap;
  flex: 0 1 100%;
}

.button {
  display: flex;
  flex: 0 1 25%;
  justify-content: center;
  align-items: center;
  height: 80px;
  border: 1px solid black;
  border-top: none;
  border-left: none;
  cursor: pointer;
  background: -webkit-linear-gradient(top, #d2d2d2, #ddd);
  font-size: 1.2em;
}

.button:nth-child(4n+1) {
  border-left: 1px solid black;
}

.button:nth-child(4n),
.button:last-child {
  background: #f98110;
  color: #fff;
}

.span-2 {
  flex-grow: 2;
}
