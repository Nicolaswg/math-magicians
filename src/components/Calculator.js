import './Calculator.css';
import React, { useState } from 'react';
import calculate from '../logic/calculate';
import Buttons from './CalButton';

const Calculator = () => {
  const [data, setData] = useState(
    {
      total: '0',
      next: null,
      operation: null,
    },
  );

  const handleClick = (e) => {
    setData((init) => (calculate(init, e.target.textContent)));
  };

  const { total, next, operation } = data;

  return (
    <div className="calculator-grid">
      <div className="output">
        {total || ''}

        {operation || ''}

        {next || ''}
      </div>
      <Buttons click={handleClick} />
    </div>
  );
};
export default Calculator;
