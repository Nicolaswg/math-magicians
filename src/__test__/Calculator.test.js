import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

it('Render Calculator correctly', () => {
  const tree = renderer.create(
    <Calculator />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
