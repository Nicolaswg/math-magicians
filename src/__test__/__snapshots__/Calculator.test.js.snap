// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Calculator correctly 1`] = `
Array [
  <h2
    className="subtitle"
  >
    Lest do some Math
  </h2>,
  <div
    className="calculator-grid"
  >
    <div
      className="output"
    >
      0
      
      
    </div>
    <div
      className="row"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="button span-2"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>,
]
`;
